### PR TITLE
Add PNG microscopy data and files for testing dataframe

### PR DESCRIPTION
**Pull request by mariehbourget**
Related to #5 
This PR aims to add microscopy data to the `data-testing` repo.
It also covers the `_sessions.tsv` and `_scans-tsv` files and architecture needed for the testing of the new loader in ivadomed.

Currently, I have an issue with the data that was already in the repo (MRI anat), because it is used for testing in ivadomed and the path is harcoded in several tests and config files:
- Ideally, we would want to have different BIDS dataset folders inside the `data-testing` repo (e.g. anat, microscopy, eeg, dwi, etc).
- So I added a microscopy BIDS folders for PNG format in `microscopy_png` and updated the `README` accordingly.
- However, files related to `MRI anat` are still at the root of data-testing (all tests passed).
- If I move them into a "mri-anat" dataset folder for example, 48 tests are failing because of the paths.

I would like to go ahead and write a test for the new loader. I can do it even if the data stays like that, but it is not very clean and I think it would be best to update the MRI anat data as well (in another PR) @lrouhier @charleygros what do you think? Thanks!

### CODE REVIEW

**Comment by lrouhier** [comment]
I think You are right this is not very clean, and even though 48 seems like a big number I don't think it would take too much time to fix (because this would only be a path_related issue). I don't know about the order though, should we break everything now or wait for the JOSS review to be over (because we will do a release after the review right ?)

**Comment by mariehbourget** [comment]
> I don't know about the order though, should we break everything now or wait for the JOSS review to be over (because we will do a release after the review right ?)

I think we should wait for the JOSS review, I'll write the tests with the current folder hierarchy of this branch and I'll fix it afterwards.

**Comment by charleygros** [comment]
> I think we should wait for the JOSS review, I'll write the tests with the current folder hierarchy of this branch and I'll fix it afterwards.

Ideally we should wait for the JOSS review. But we don't really know when it will finish (only one reviewer left tho): I don't want to freeze your development for an uncertain period of time :/

**Comment by mariehbourget** [comment]
> Ideally we should wait for the JOSS review. But we don't really know when it will finish (only one reviewer left tho): I don't want to freeze your development for an uncertain period of time :/

I wrote the tests for the function with the file architecture of this branch in PR [#584](https://github.com/ivadomed/ivadomed/pull/584) in `ivadomed` repo.
All previous tests are still passing on ivadomed.

Here is what I suggest:
1. Merge this PR
2. Create a new release for `data-testing`
3. Update the link in `download_data` in PR #584 in ivadomed [here](https://github.com/ivadomed/ivadomed/blob/5564e6345a0bffdee52631cae4f9d6f0fe50c122/ivadomed/scripts/download_data.py#L25) to point to the new `data-testing` release
4. Merge PR 584 in ivadomed (all tests should still passed)
5. Wait for JOSS review
6. Open a new PR here to update files architecture, and a new PR on ivadomed to fix paths in tests.

If we want to be hyper-cautious, we can wait for JOSS review before step 4. This won't freeze the development on my part.

**Comment by jcohenadad** [comment]
> If we want to be hyper-cautious, we can wait for JOSS review before step 4. This won't freeze the development on my part.

I think it's fine. Worst case scenario: there is a glitch for a few hours, while one reviewer is looking at the repos, but they would easily understand it's a temporary issue that will be fixed a few hours later. If it glitches happen to Google it can happen to us 😉

**Comment by mariehbourget** [comment]
This PR is ready for review.
After merged, a new release must be done in order to update the download link in [#584](https://github.com/ivadomed/ivadomed/pull/584) (ivadomed).
@charleygros or @andreanne-lemay, would you like to review?

File: microscopy_png/participants.tsv
Comment (andreanne-lemay) [inline microscopy_png/participants.tsv:2]: Will the splits be done on participants or samples? If the split is done on participants, having 3 subjects can be convenient (1 for training, 1 for validation, and 1 for testing). 2 participants will work, it will only be necessary to change the parameter "test_fraction": 0.0, when/if testing training with microscopy data.

**Review by andreanne-lemay — APPROVED** [review]
Looks great, thank you!

File: microscopy_png/participants.tsv
Comment (mariehbourget) [reply microscopy_png/participants.tsv:2]: Split could be done on either subject or sample. I tried to keep the dataset as light as possible. I think we should duplicate files in the tests if we need more subjects (as it is done with MRI).

Comment (andreanne-lemay) [reply microscopy_png/participants.tsv:2]: In this line of thought, why not create the dataset with a single patient as it is done with MRI?

Comment (mariehbourget) [reply microscopy_png/participants.tsv:2]: I wanted to add files for the "sessions" and "scans" architectures. It seemed clearer to me to do it directly on the dataset than "buried" in a test somewhere.  I made sure than the dataset is under 3Mb though.

File: microscopy_png/derivatives/dataset_description.json
Comment (charleygros) [inline microscopy_png/derivatives/dataset_description.json]:
```suggestion
{
    "Name": "Example derivative dataset",
    "BIDSVersion": "1.0.2",
    "PipelineDescription": {"Name": "Example derivative pipeline"}
}
```

**Review by charleygros — APPROVED** [review]

File: microscopy_png/derivatives/dataset_description.json
Comment (mariehbourget) [reply microscopy_png/derivatives/dataset_description.json]: Fixed in d62b405, thanks!

**Comment by mariehbourget** [comment]
Thank you @andreanne-lemay and @charleygros, I'll merge and create the new release.

**Pull request merged**
